staging/src/k8s.io/apimachinery/pkg/runtime/schema/group_version.go

```diff
@@ -303,3 +305,12 @@ func FromAPIVersionAndKind(apiVersion, kind string) GroupVersionKind {
 	}
 	return GroupVersionKind{Kind: kind}
 }
+
+type APILifecycle struct {
+	IntroducedVersion *version.Version
```

Comment:
The pointer here is used to indicate that the value is optional? Should this be optional?  Or should we pick a version number to use to indicate epoch (APIs that have been around ~forever)?

Reply:
I think it should be optional, because most APIs right now does not have this specified.

Reply:
Making it optional for fallback purposes makes sense.  Can we document that here?  It's probably also worth mentioning what a nil value means in both cases.  For `IntroducedVersion`, it sounds like we're going with "it's been around long enough that we don't track the introduced version" and for `RemovedVersion` is means "it' never been assigned a removal version`.

Reply:
Added documents.